### PR TITLE
Storage points tracking refactoring

### PR DESCRIPTION
**Pull request by generall**

### CODE REVIEW

File: lib/api/src/grpc/proto/collections.proto [hunk 1]
```diff
@@ -180,6 +180,7 @@ message CollectionInfo {
   uint64 ram_data_size = 6; // Used RAM (not implemented)
   CollectionConfig config = 7; // Configuration
   map<string, PayloadSchemaInfo> payload_schema = 8; // Collection data types
+  uint64 points_count = 9; // number of vectors in the collection
```
Comment (IvanPleshkov) [inline lib/api/src/grpc/proto/collections.proto:183]: what is difference between `CollectionInfo.vectors_count`? I didn't catch main idea

File: lib/segment/src/index/struct_payload_index.rs [hunk 2]
```diff
@@ -338,12 +334,14 @@ impl PayloadIndex for StructPayloadIndex {
...
-            let points_iterator_ref = self.points_iterator.borrow();
+            let points_iterator_ref = self.id_tracker.borrow();
             let struct_filtered_context = self.struct_filtered_context(query);
 
             // CPU-optimized strategy here: points are made unique before applying other filters.
             // ToDo: Implement iterator which holds the `visited_pool` and borrowed `vector_storage_ref` to prevent `preselected` array creation
-            let mut visited_list = self.visited_pool.get(points_iterator_ref.max_id() as usize);
+            let mut visited_list = self
+                .visited_pool
+                .get(points_iterator_ref.max_id() as usize + 1);
```
Comment (e-ivkov) [inline lib/segment/src/index/struct_payload_index.rs:344]: Is this a bug fix? I can't notice any other logic changes

Comment (generall) [reply lib/segment/src/index/struct_payload_index.rs:344]: It is not a bug, cause visited_list is able to auto-adjust. But should give some perf. improvement

File: lib/collection/src/collection_manager/holders/proxy_segment.rs [hunk 3]
```diff
@@ -373,7 +373,8 @@ impl SegmentEntry for ProxySegment {
 
         SegmentInfo {
             segment_type: SegmentType::Special,
-            num_vectors: self.vectors_count(),
+            num_vectors: self.points_count(), // ToDo: account number of vector storages
```
[the remark this answers is not in the document]
Comment (generall) [reply lib/collection/src/collection_manager/holders/proxy_segment.rs:376]: made as a preparation for https://github.com/qdrant/qdrant/issues/345

**Review by e-ivkov — APPROVED** [review]

**Pull request merged**